src/notedata.h

```diff
@@ -4,6 +4,15 @@
 #include <QObject>
 #include <QDateTime>
 
+struct NoteExport
+{
+    QString id;
+    QString fullTitle;
+    QDateTime lastModificationDateTime;
+    QDateTime creationDateTime;
+    QString content;
+};
+
```

Comment:
There might be confusion between them in code. 

NoteData is in QObject to use Qt deletion mechanism. I think there is memory leaks in your code, there is no delete after creating noteExport dynamically.